File: generator/src/main/java/com/github/payne/generator/JAVA/GeneratorModel.java
package com.github.payne.generator.JAVA;

import com.github.payne.generator.JAVA.model.BasicData;
import com.github.payne.generator.JAVA.model.LanguagesData;
import lombok.Data;
import lombok.NonNull;

@Data
public class GeneratorModel {

    @NonNull
    private final BasicData basicData;
    {
        LanguagesData f = new LanguagesData();
    }
}

File: generator/src/main/java/com/github/payne/generator/GeneratorModel.kt
package com.github.payne.generator


class GeneratorModel {
    val properties = mutableMapOf(
            "org.gradle.daemon" to "false", // was having very bad memory usage with daemon+Android
            "org.gradle.jvmargs" to "-Xms512M -Xmx4G -XX:MaxPermSize=1G -XX:MaxMetaspaceSize=1G",
            "org.gradle.configureondemand" to "false")
}